modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/utils/sortItems.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {IOrderable} from './types';
+
+export default function sortItems(
```

Comment:
I'd personally like seeing documentation added to this method about why we have these `ordered` and `notOrdered` management to avoid making multiple requests to the FDSView and FDSField/Filter/Sort/Action. 

Seeing this at first was strange why the `fdsFiltersOrder`/`fdsSortsOrder` wasn't completely in sync with the actual saved FDSFilters/FDSSorts data.

Reply:
yes, I agree this is odd. An interesting point of discussion is to consider that, before this change, the order was neither in sync and that caused issues. Basically, this leads to the fact that current solution sets a convention that produces a consistent result in different use cases, with less effort in terms of persistence (thus preventing complex transactions), at the cost of a potential minor data inconsistency in the persistence.

Also, note that our headless APIs for FDSView and friends can always be invoked to create the DSM entities making up a Data Set, meaning that to some extent system must consider data inconsistencies as not everything can be validated in the backend. This is a broader topic where fragment has a lot to say